src/common/indexer/elastic/elastic.indexer.service.ts

```diff
@@ -856,25 +857,45 @@ export class ElasticIndexerService implements IndexerInterface {
   }
 
   async setAccountAssetsFields(address: string, assets: AccountAssets): Promise<void> {
-    return await this.elasticService.setCustomValues('accounts', address, {
-      assets: { ...assets },
-    });
+    return await this.elasticService.setCustomValues('accounts', address, { assets });
+  }
+
+  async ensureAccountsWritable(): Promise<void> {
+    await this.ensureCollectionWritable('accounts');
   }
 
-  async setAccountExtraFields(address: string, txCount: number, scrCount: number, deployedAt: number | null, txCount24h: number, txCount7d: number, txCount30d: number): Promise<void> {
-    const updateExtraFields: any = {
-      txCount: txCount,
-      scrCount: scrCount,
-      txCount24h: txCount24h,
-      txCount7d: txCount7d,
-      txCount30d: txCount30d,
-    };
+  async ensureTokensWritable(): Promise<void> {
+    await this.ensureCollectionWritable('tokens');
+  }
+
+  private async ensureCollectionWritable(collection: string) {
+    const query = new ElasticQuery().withPagination({ from: 0, size: 1 });
+    const items = await this.elasticService.getList(collection, 'id', query);
+
+    if (items.length === 0) {
+      throw new Error(`No entries available in the '${collection}' collection`);
+    }
+
+    const item = items[0];
 
-    if (deployedAt !== null) {
-      updateExtraFields.deployedAt = deployedAt;
+    try {
+      await this.elasticService.setCustomValue(collection, item.id, 'ensureWritable', undefined);
```

Comment:
we should delete this at the end because there are scripts (elastic-reindexer-go for example) that will copy all the records from a cluster to another and it can fail if there are such entries that don't follow the index mapping

Reply:
it normally should not create any new data in the record, it just ensures that it can write a dummy value and write rights to the collection are available